src/applications/pre-need/definitions/address.js

```diff
@@ -0,0 +1,228 @@
+    && typeof address.postalCode !== 'undefined';
+
+  if (hasAddressInfo && typeof address.state === 'undefined') {
+    errors.state.addError('Please enter a state or province, or remove other address information.');
+  }
+
+  validatePostalCodes(errors, address);
+}
+
+const countryValues = countries.map(object => object.value);
+const countryNames = countries.map(object => object.label);
+const militaryStates = states.USA
+  .filter(state => state.value === 'AE' || state.value === 'AP' || state.value === 'AA')
+  .map(state => state.value);
+const militaryLabels = states.USA
+  .filter(state => state.value === 'AE' || state.value === 'AP' || state.value === 'AA')
+  .map(state => state.label);
+const usaStates = states.USA.map(state => state.value);
+const usaLabels = states.USA.map(state => state.label);
+const canProvinces = states.CAN.map(state => state.value);
+const canLabels = states.CAN.map(state => state.label);
+
+function isMilitaryCity(city = '') {
+  const lowerCity = city.toLowerCase().trim();
+
+  return lowerCity === 'apo' || lowerCity === 'fpo' || lowerCity === 'dpo';
+}
+
+const requiredFields = ['street', 'city', 'country', 'state', 'postalCode'];
+
+/*
+ * Create schema for addresses
+ *
+ * @param {object} schema - Schema for a full form, including address definition in definitions
+ * @param {boolean} isRequired - If the address is required or not, defaults to false
+ */
+export function schema(currentSchema, isRequired = false) {
+  const addressSchema = currentSchema.definitions.address;
+  return {
+    type: 'object',
+    required: isRequired ? requiredFields : [],
+    properties: _.assign(addressSchema.properties, {
+      country: {
+        'default': 'USA',
+        type: 'string',
+        'enum': countryValues,
+        enumNames: countryNames
+      },
+      state: {
+        title: 'State',
+        type: 'string',
+        maxLength: 3
+      },
+      postalCode: {
+        type: 'string',
+        maxLength: 6
+      }
+    })
+  };
+}
+
+/*
+ * Create uiSchema for addresses
+ *
+ * @param {string} label - Block label for the address
+ * @param {boolean} useStreet3 - Show a third line in the address
+ * @param {function} isRequired - A function for conditionally setting if an address is required.
+ *   Receives formData and an index (if in an array item)
+ * @param {boolean} ignoreRequired - Ignore the required fields array, to avoid overwriting form specific
+ *   customizations
+ */
+export function uiSchema(label = 'Address', useStreet3 = false, isRequired = null, ignoreRequired = false) {
+  let fieldOrder = ['country', 'street', 'street2', 'street3', 'city', 'state', 'postalCode'];
+  if (!useStreet3) {
+    fieldOrder = fieldOrder.filter(field => field !== 'street3');
+  }
+
+  const addressChangeSelector = createSelector(
+    ({ formData, path }) => _.get(path.concat('country'), formData),
+    ({ formData, path }) => _.get(path.concat('city'), formData),
+    _.get('addressSchema'),
```

Comment:
how does this work with one argument?

Reply:
Lodash/fp does auto-currying, so this returns a function that takes formData and gets the object at `addressSchema`. I've changed my mind about using that feature since this code was originally written, so I think I'll change it.

Reply:
I thought that might be the case but it's not being used as a function in the `resultFunc` e.g. `addressSchema.properties` on line 121.